site/public/js/gradeable-notebook.js

```diff
@@ -79,13 +79,49 @@ function notebookAutosaveKey() {
  * Saves the current state of the notebook gradeable to localstorage.
  */
 function saveNotebookToLocal() {
-    if (typeof autosaveEnabled !== 'undefined' && autosaveEnabled) {
-        localStorage.setItem(notebookAutosaveKey(), JSON.stringify({
-            timestamp: Date.now(),
-            multiple_choice: gatherInputAnswersByType('multiple_choice'),
-            codebox: gatherInputAnswersByType('codebox'),
-        }));
-    }
+    const mc_inputs=[];
+    const short_answer_inputs = [];
+
+    //loop through multiple choice questions and save answers
+    $('.multiple_choice').each(function(){
+        let file_name='';
+        $(this).children('fieldset').each(function(){
+            file_name = $(this).data('filename');
+            if (file_name) {
+                const answers = [];
+                $(this).children('label').each(function(){
+                    //grab selected answers
+                    $(this).children('input').each(function(){
+                        if ($(this)[0].checked) {
+                            answers.push($(this)[0].defaultValue);
+                        }
+                    });
+                });
+                mc_inputs.push([file_name, answers]);
```

Comment:
I would recommend using an object to store the filename and value instead of an array. As someone taking a look at this code for the first time, I was confused about what `inputs.multiple_choice[id][0]` was vs `inputs.multiple_choice[id][1]`. In javascript the keys will be implied if not specified, so if you do
```suggestion
                mc_inputs.push({file_name, answers});
```
then this would make an object like this:
```
{
    file_name: file_name,
    answers: answers
}
```
And then when loading, you can use the object instead of the array for more clarity